### PR TITLE
Add support for logs data

### LINKED ISSUE #331: Support for Logs in Grafana
Support for the "Logs" feature.

https://grafana.com/docs/grafana/latest/explore/logs-integration/
https://grafana.com/docs/grafana/latest/developers/plugins/build-a-logs-data-source-plugin/

Reason:

I am using ClickHouse as a log storage system and would like to be able to use the logs feature. I have a table with about over 300 columns for normalizing the logs fields, a column for the original log message, and a column for the log timestamp. For each row many of the normalization columns are NULL because many of the logs do not use all the fields so I was thinking that the Logs feature would work well for this.

Suggestion:

The idea is I could map the columns to the fields required for the Logs feature like timestamp and the log message and than for the other fields in each row add any non-null values as additional fields in the result.

Would this be something you would consider? I am willing to help just need some help on where I need to modify because I am not familiar with Grafana data source plugins.

### PR DESCRIPTION
**Pull request by pixelsquared**
fixes #331 

This is my attempt at adding support for logs to clickhouse-grafana.

I have not added any tests yet.

### CODE REVIEW

**Comment by pixelsquared** [comment]
@Slach Could you review this and tell me what you think?

**Review by Slach — CHANGES_REQUESTED** [review]
thanks a lot for your efforts, did you try to run `docker-compose run --rm frontend_builder` in your branch?
currently, I see the following error:
```
ERROR in module.js from UglifyJs
Unexpected token: keyword «const» [../node_modules/apache-arrow/type.js:19,0][module.js:4960,0]
npm ERR! code ELIFECYCLE
npm ERR! errno 2
```

**Comment by pixelsquared** [comment]
> thanks a lot for your efforts, did you try to run `docker-compose run --rm frontend_builder` in your branch?
> currently, I see the following error:
> 
> ```
> ERROR in module.js from UglifyJs
> Unexpected token: keyword «const» [../node_modules/apache-arrow/type.js:19,0][module.js:4960,0]
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> ```

I am not getting that error when I run the frontend_builder. What command did you run to get that error?

File: src/sql_series.ts [hunk 1]
```diff
@@ -163,6 +189,45 @@ export default class SqlSeries {
...
+            case 'Nullable(Int64)':
+            case 'Nullable(Float32)':
+            case 'Nullable(Float64)':
+            case 'Nullable(Decimal)':
+            case 'Nullable(Decimal32)':
+            case 'Nullable(Decimal64)':
+            case 'Nullable(Decimal128)':
+                return FieldType.number;
+            case 'DateTime':
+            case 'DateTime64':
```
[the remark this answers is not in the document]
Comment (pixelsquared) [reply src/sql_series.ts:224]: Added the additional types.

**Comment by Slach** [comment]
> I am not getting that error when I run the frontend_builder. What command did you run to get that error?
Sorry, I know nothing, how you tested your changes
just simple steps to reproduce
```
git clone https://github.com/pixelsquared/clickhouse-grafana.git ./clickhouse-grafana-logs
cd ./clickhouse-grafana-logs
docker-compose run --rm frontend_builder
```
but I tied it from scratch now, and it doesn't reproduce anymore
nice ;)
after 
```
docker-compose run --rm plugin_signer
docker-compose up -d clickhouse grafana
```

I open `http://localhost:3000` and see plugin loaded
but when I try to create a new dashboard
I got the following error
image
Also when I try to test datasources
I got same error
image
and in Console i see following stacktrace
image

I tried `GRAFANA_VERSION=7.4.3 docker-compose up -d grafana` and `GRAFANA_VERSION=7.5.2 docker-compose up -d grafana`, same result

**Review by Slach — CHANGES_REQUESTED** [review]
could you please provide a functinal tests and documentation with screenshots?

**Comment by pixelsquared** [comment]
Sorry some stuff came up and I will not have time to work on this until next week. I will add tests I just wanted to make sure I was going in the right direction first. I am not sure what is causing those error you are seeing I will work on it next week.

**Comment by Slach** [comment]
@pixelsquared any news and progress on this PR?

**Comment by pixelsquared** [comment]
Sorry about the lack of response I got busy with some stuff but I am back to working on this now.

**Comment by pixelsquared** [comment]
For screenshots in the documentation where should I put the images?

**Comment by Slach** [comment]
we use GitHub user content
just copy the image to buffer
and paste it via CTRL+V (cmd+V) inside text area for comments in this PR
you will get text link something like 
```
![image](https://user-images.githubusercontent.com/105560/114646031-f10bcb00-9cf3-11eb-8832-3f0844c4acc6.png)
```
if you copy and paste it to README.md you will get something like
after markdown formatting

image

**Comment by Slach** [comment]
@pixelsquared any progress?
don't give up ;) feel free to ask anything, I try to help you

**Comment by medvetskiy** [comment]
Hi guys,
do you have any progress?
it is very useful feature for making cool dashboards.

**Comment by Slach** [comment]
no progress more than 2 month
@pixelsquared will you continue efforts?

**Pull request closed without merging**